src/odemis/gui/cont/tabs.py

```diff
@@ -3255,10 +3255,33 @@ def _display_meteor_pos_warning_msg(self, end_pos) -> bool:
                 else:
                     pos_str.append(f"{axis} = " + readable_str(end_pos[axis], "m", 4))
         pos_str = "\n". join(pos_str)
-        box = wx.MessageDialog(self.main_frame,
-                               "The stage will move to this position:\n" + pos_str + "\n\nIs this safe?",
-                               caption="Large move of the stage",
-                               style=wx.YES_NO | wx.ICON_QUESTION | wx.CENTER)
+
+        # Check the deviation in rotation angle when
+        # switching from SEM to FM,
+        # give a warning message if switching is done from a different rotation angle
+        target_label = self.tab_data_model.main.posture_manager.getCurrentPostureLabel(end_pos)
+        current_angle = 0
+        fav_angle = 0
+        angle = ''
+        if target_label == FM_IMAGING:
+            stage_md = self._stage.getMetadata()
+            fav_angles = stage_md[model.MD_FAV_SEM_POS_ACTIVE]
+            key = "rz" if "rz" in fav_angles else "rm"
+            current_angle = self._stage.position.value[key]
+            fav_angle = fav_angles[key]
+
+        if not math.isclose(current_angle, fav_angle) and target_label == FM_IMAGING:
+            box = wx.MessageDialog(self.main_frame,
+                                   "The current " + key + " value is different from the desired value. "
+                                   "The switching behavior may not be proper." + "\n\nThe stage will move to "
+                                   "this position:\n" + pos_str + "\n\n Is it safe?",
+                                   caption="Large move of the stage",
+                                   style=wx.YES_NO | wx.ICON_QUESTION | wx.CENTER)
```

Comment:
I think it's nicer to use the same structure as in the plugin: in the `if` block, set a `warn_msg` string, dependent on the angle, and after the block, do one `box = wx.MessageDialog(self.main_frame, warn_msg,...)`.